frontend/components/cards/ModuleUnionCard.tsx

```diff
@@ -22,22 +23,52 @@ const ModuleUnionCard: FC<ModuleUnionCardProps> = ({
       content = <HTMLEllipsis unsafeHTML={module.captionHtml} maxLine="3" basedOn="words" />;
       break;
     case "propositions.occurrence":
-      content = <div dangerouslySetInnerHTML={{ __html: module.title }} />;
+      content = module.meta &&
+        module.meta.highlight &&
+        module.meta.highlight.text &&
+        module.meta.highlight.text.length > 0 && ( // TODO: Replace Regex with more formal fix
+          <Box sx={{ "& *": { fontSize: "0.8rem", backgroundColor: "rgba(0, 0, 0, 0)" } }}>
+            <HTMLEllipsis
+              unsafeHTML={module.meta.highlight.text[0].replace(/(<([^>]+)>)/gi, "")}
+              maxLine="4"
+              basedOn="words"
+            />
+          </Box>
+        );
       break;
     case "propositions.proposition":
-      content = <div dangerouslySetInnerHTML={{ __html: module.title }} />;
+      content = module.meta &&
+        module.meta.highlight &&
+        module.meta.highlight.text &&
+        module.meta.highlight.text.length > 0 && ( // TODO: Replace Regex with more formal fix
+          <Box sx={{ "& *": { fontSize: "0.8rem", backgroundColor: "rgba(0, 0, 0, 0)" } }}>
+            <HTMLEllipsis
+              unsafeHTML={module.meta.highlight.text[0].replace(/(<([^>]+)>)/gi, "")}
+              maxLine="4"
+              basedOn="words"
+            />
+          </Box>
+        );
       break;
     case "quotes.quote": {
       content = (
-        <blockquote className="blockquote">
-          <HTMLEllipsis unsafeHTML={module.bite} maxLine="4" basedOn="words" />
-          {module.attributeeString && (
-            <footer
-              className="blockquote-footer"
-              dangerouslySetInnerHTML={{ __html: module.attributeeString }}
-            />
-          )}
-        </blockquote>
+        //<blockquote className="blockquote">
+        <div>
+          {(module.meta &&
+            module.meta.highlight &&
+            module.meta.highlight.text &&
+            module.meta.highlight.text.length > 0 && ( // TODO: Replace Regex with more formal fix
+              <Box sx={{ "& *": { fontSize: "0.8rem", backgroundColor: "rgba(0, 0, 0, 0)" } }}>
+                <HTMLEllipsis
+                  unsafeHTML={module.meta.highlight.text[0].replace(/(<([^>]+)>)/gi, "")}
```

Comment:
It looks like the regex pattern here removes the HTML tags but leaves their contents, which _mostly_ works but unfortunately has weird results in some cases. For example, if there's a header tag captured in the highlight, simply stripping out all the HTML tags results in something like this (notice the "Adam See" at the beginning):

![adam see](https://user-images.githubusercontent.com/25336795/132631322-e0246d52-49eb-4a7a-9734-2c4806b36dc2.png)

It also would be nice to do something with the <mark> tags (which contains the text that matched the search). Since highlighting doesn't look good on the card, maybe we could apply a **bold** style instead?

^ So, considering both of these things, I think we should:
- Change the regex pattern to remove header tags _and their contents_ (i.e., in the example above, remove the initial "Adam" altogether) but leave other HTML tags present
- Add an additional style rule for `mark` tags to make them bold—like this:
```
sx={{ "& *": { fontSize: "0.8rem", backgroundColor: "rgba(0, 0, 0, 0)" }, "& mark": { fontWeight: "bold" } }}
```

Reply:
Oh my mistake, I didn't know that's what you meant. I'll update the regexs!